src/tiled/automapper.cpp

```diff
@@ -691,7 +691,7 @@ static bool compareLayerTo(const TileLayer *setLayer,
                 bool matchListNo  = false;
 
 
-                if (!setLayer->contains(x + offset.x(), y + offset.y())) {
+                if (!(setLayer->contains(x + offset.x(), y + offset.y()) || setLayer->map()->infinite())) {
```

Comment:
Intuitively, this check and the special handling inside should actually not be necessary at all. If the `setLayer` does not contain the given location, then `c1` will just be an empty cell, which I think should be treated the same as a cell outside of the layer boundaries. Can you check this?